### PR TITLE
Remove GT_ARGPLACE nodes

### PR DESCRIPTION
**Pull request by jakobbotsch**
These do not serve much purpose today -- instead just use null and add a
helper function to iterate non-null early args, which is somewhat
common.

In addition to saving some TP and memory, teaching the backend about
null early nodes will also be beneficial because I am planning to change
rationalization to null out non-values in the early arg list so that all
nodes have only values as their operands in LIR.

[No diffs and some nice throughput gains:](https://dev.azure.com/dnceng/public/_build/results?buildId=1734149&view=ms.vss-build-web.run-extensions-tab)
<html>
<body>

Collection | Base # instructions | Diff # instructions | PDIFF
-- | -- | -- | --
aspnet.run.windows.x64.checked.mch | 69,717,468,395 | 69,206,312,087 | -0.73%
benchmarks.run.windows.x64.checked.mch | 54,695,846,729 | 54,294,078,768 | -0.73%
coreclr_tests.pmi.windows.x64.checked.mch | 340,169,515,528 | 337,478,749,067 | -0.79%
libraries.crossgen2.windows.x64.checked.mch | 128,653,906,043 | 126,926,566,191 | -1.34%
libraries.pmi.windows.x64.checked.mch | 228,653,702,806 | 226,554,618,843 | -0.92%
libraries_tests.pmi.windows.x64.checked.mch | 531,053,530,645 | 525,233,144,101 | -1.10%

</body>
</html>

Memory stats (libraries.pmi)
https://www.diffchecker.com/0iigCXGB
Before: 25961399533 bytes
After: 25770612141 bytes (-0.7%)

### CODE REVIEW

**Comment by jakobbotsch** [comment]
cc @dotnet/jit-contrib, any thoughts and does anyone want to take this one?

File: src/coreclr/jit/morph.cpp [hunk 1]
```diff
@@ -8352,33 +8348,30 @@ void Compiler::fgMorphRecursiveFastTailCallIntoLoop(BasicBlock* block, GenTreeCa
     // Process early args. They may contain both setup statements for late args and actual args.
     // Early args don't include 'this' arg. We need to account for that so that the call to gtArgEntryByArgNum
     // below has the correct second argument.
-    for (CallArg& arg : recursiveTailCall->gtArgs.Args())
+    for (CallArg& arg : recursiveTailCall->gtArgs.EarlyArgs())
```
Comment (AndyAyersMS) [inline src/coreclr/jit/morph.cpp:8353]: Is the comment above stale?

**Review by AndyAyersMS — APPROVED** [review]
Thanks. Another good cleanup.

File: src/coreclr/jit/morph.cpp (hunk 1 above)
Comment (jakobbotsch) [reply src/coreclr/jit/morph.cpp:8353]: Yep, will fix that.

File: src/coreclr/jit/flowgraph.cpp [hunk 2]
```diff
@@ -3887,12 +3887,6 @@ GenTree* Compiler::fgSetTreeSeq(GenTree* tree, bool isLIR)
             if (m_isLIR)
             {
                 node->ClearReverseOp();
-
-                // ARGPLACE nodes are not threaded into the LIR sequence.
```
Comment (SingleAccretion) [inline src/coreclr/jit/flowgraph.cpp:3891]: It's probably not worth re-running the CI for this, but the function header now has an outdated comment:
```
//    isLIR - whether the sequencing is being done for LIR. If so,
//            ARGPLACE nodes will not be threaded into the linear
//            order, and the GTF_REVERSE_OPS flag will be cleared
```

Comment (jakobbotsch) [reply src/coreclr/jit/flowgraph.cpp:3891]: There was another instance of this and I also took the opportunity to change some uses of `GetEarlyNode()` to `GetNode()`.

**Pull request merged**